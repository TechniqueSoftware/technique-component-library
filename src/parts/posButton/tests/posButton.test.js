import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import PosButton from '..'

describe('<PosButton>', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<PosButton />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders', () => {
    render(<PosButton />)
    expect(screen.getByLabelText('pos order')).toBeInTheDocument()
  })

  it('button click', () => {
    render(<PosButton />)
    expect(screen.getByLabelText('pos order')).toBeInTheDocument()
    userEvent.click(screen.getByLabelText('pos order'))
    expect(localStorage.getItem('pos')).toBe('1')
  })
})
