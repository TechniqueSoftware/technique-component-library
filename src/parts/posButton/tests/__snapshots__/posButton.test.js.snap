// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PosButton> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="posButtonContainerMargin"
  >
    <button
      class="
    d-flex
    align-items-center
    justify-content-center
    flex-row
    button
    
    undefined
   btn btn-link text-primary btn-md"
      role="button"
      title="Point of Sale"
      type="button"
    >
      <svg
        aria-label="pos order"
        fill="none"
        height="29"
        width="33"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 23.27 3.705 9.207c.215-1.773 1.534-1.705 2.387-1.705 6.904-.028 20.952-.068 21.906 0 .955.068 1.364 1.165 1.45 1.705l1.619 14.065c.028.426 0 1.497 0 2.386 0 1.279-.341 1.281-1.023 1.279-8.297-.029-25.945 0-27.106 0-.512 0-.938 0-.938-.512v-3.153Z"
          fill="#F9FAFF"
        />
        <path
          d="m3.705 9.206 1.24.15-1.24-.15ZM2 23.271l-1.24-.15-.01.074v.076H2Zm28.044 3.665-.004 1.25.004-1.25Zm1.023-3.665 1.247-.084-.002-.03-.003-.03-1.242.144Zm-1.62-14.065 1.242-.143-.003-.026-.004-.026-1.235.195Zm-1.449-1.705-.089 1.247.09-1.247Zm-21.906 0v1.25h.005l-.005-1.25ZM2.464 9.056.759 23.12l2.482.301L4.946 9.356l-2.482-.3ZM.75 23.27v3.153h2.5v-3.153H.75Zm0 3.153c0 .299.063.657.285.99.227.34.536.528.803.628.431.162.931.144 1.1.144v-2.5c-.14 0-.215 0-.274-.005-.062-.004-.026-.009.052.02a.851.851 0 0 1 .399.326.75.75 0 0 1 .135.397H.75Zm2.188 1.762c1.166 0 18.808-.029 27.102 0l.008-2.5c-8.299-.029-25.954 0-27.11 0v2.5Zm27.102 0c.142 0 .398.004.643-.034.275-.043.652-.152.984-.464a1.97 1.97 0 0 0 .552-1.003 4.52 4.52 0 0 0 .098-1.028h-2.5c0 .134-.005.236-.011.314a1.039 1.039 0 0 1-.02.138c-.008.037.016-.099.17-.243a.77.77 0 0 1 .225-.148.524.524 0 0 1 .119-.036l-.055.003a5.579 5.579 0 0 1-.197 0l-.008 2.5Zm2.277-2.529c0-.804.03-1.99-.003-2.47l-2.494.167c.024.372-.003 1.33-.003 2.303h2.5Zm-.008-2.53-1.62-14.064-2.483.286 1.62 14.065 2.483-.286ZM30.682 9.012a3.996 3.996 0 0 0-.591-1.525c-.36-.542-1.005-1.16-2.004-1.232l-.178 2.494-.007-.002c-.006-.002-.006-.003 0 0a.46.46 0 0 1 .105.122c.051.076.099.173.137.28.039.107.06.2.069.253l2.47-.39Zm-2.595-2.757c-.537-.038-4.364-.044-8.812-.04-4.496.006-9.736.023-13.189.037l.01 2.5c3.452-.014 8.689-.031 13.181-.036 4.54-.005 8.215.003 8.632.033l.178-2.494ZM6.092 6.251c-.334 0-1.18-.04-1.938.294a2.671 2.671 0 0 0-1.135.938c-.307.446-.483.978-.555 1.573l2.482.3c.035-.292.105-.416.132-.456a.175.175 0 0 1 .086-.068c.06-.026.162-.054.333-.068.179-.016.351-.013.595-.013v-2.5Z"
          fill="#F37120"
        />
        <path
          d="M7.882 5.455h4.603v1.364H7.882z"
          fill="#F9FAFF"
          stroke="#F37120"
          stroke-width="1.364"
        />
        <mask
          fill="#fff"
          id="a"
        >
          <rect
            height="5.114"
            rx="0.614"
            width="8.695"
            x="5.836"
          />
        </mask>
        <rect
          fill="#F37120"
          height="5.114"
          mask="url(#a)"
          rx="0.614"
          stroke="#F37120"
          stroke-width="5"
          width="8.695"
          x="5.836"
        />
        <mask
          fill="#fff"
          id="b"
        >
          <rect
            height="14.832"
            rx="0.614"
            width="6.819"
            x="19.474"
            y="0.17"
          />
        </mask>
        <rect
          fill="#F9FAFF"
          height="14.832"
          mask="url(#b)"
          rx="0.614"
          stroke="#F37120"
          stroke-width="5"
          width="6.819"
          x="19.474"
          y="0.17"
        />
        <mask
          fill="#fff"
          id="c"
        >
          <rect
            height="3.41"
            rx="0.614"
            width="7.672"
            x="7"
            y="10.058"
          />
        </mask>
        <rect
          fill="#F9FAFF"
          height="3.41"
          mask="url(#c)"
          rx="0.614"
          stroke="#F37120"
          stroke-width="3.41"
          width="7.672"
          x="7"
          y="10.058"
        />
        <circle
          cx="13"
          cy="16"
          fill="#F37120"
          r="0.5"
          stroke="#F37120"
        />
        <circle
          cx="9"
          cy="16"
          fill="#F37120"
          r="0.5"
          stroke="#F37120"
        />
        <path
          d="M2.085 21.935h28.982M17.428 13.752h10.911"
          stroke="#F37120"
          stroke-width="2.5"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
